src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -541,6 +541,67 @@ public function dumpFile($filename, $content, $mode = 0666)
     }
 
     /**
+     * Returns the "chmod" of the file or directory
+     *
+     * @param string $path A file or directory path.
+     * @param boolean $groups Get file|directory permissions by groups ex: 'u-wxrw--wt'.
+     *                        formats:
+     *                        s => socket;
+     *                        l => symbolic link;
+     *                        - => regular;
+     *                        b => block special;
+     *                        d => directory;
+     *                        c => character special;
+     *                        p => FIFO pipe;
+     *                        u => unknown.
+     *
+     * @return string Permissions of the file|directory or throw an exception on failure.
+     */
+    public function permissions($path, $groups = false)
+    {
```

Comment:
In the `Filesystem` class we always mimic the names of the equivalent PHP functions. So this new function "must" be called [fileperms](http://php.net/fileperms) instead. Using the same names reduces the learning curve.